third_party/move/move-core/types/src/value.rs

```diff
@@ -604,9 +595,10 @@ impl TryInto<TypeTag> for &MoveTypeLayout {
             MoveTypeLayout::Signer => TypeTag::Signer,
             MoveTypeLayout::Vector(v) => TypeTag::Vector(Box::new(v.as_ref().try_into()?)),
             MoveTypeLayout::Struct(v) => TypeTag::Struct(Box::new(v.try_into()?)),
-            MoveTypeLayout::Tagged(tag, v) => match tag {
-                LayoutTag::IdentifierMapping(_) => v.as_ref().try_into()?,
-            },
+
+            // Native layout variant is only used at runtime, and so we should
+            // not be able to construct a type tag from it here.
+            MoveTypeLayout::Native(..) => bail!("No type tag exists for native layout"),
```

Comment:
Why are we guaranteed to never compute type tag of a layout during runtime?

Reply:
So we have multiple value representations, `Value` inside the VM and `MoveValue` defined here. The first one differs from the second because it is mutable (via interior mutability). 

Personally, I don't like the fact that we have so many different types for the same thing (it applies to values, types, etc.) but it is what it is😄 the native layouts are only needed for VM, so there is no need to have it for `MoveValue` and it is better to error.